### PR TITLE
feat(ask and expert): successfully notify user when ask an expert

### PR DESCRIPTION
**Pull request by dilin-MS**
1. feat(ask and expert): successfully notify user when ask an expert
2. Update function and file comments.
3. Update manifest file name.

### CODE REVIEW

File: faq-plus/bot/common/adaptiveHelper.ts [hunk 1]
```diff
@@ -27,11 +34,7 @@ export async function askAnExpertSubmitText(
       attachments: [getAskAnExpertCard(askAnExpertSubmitTextPayload)],
     } as Activity;
 
-    try {
-      await turnContext.updateActivity(updateCardActivity);
-    } catch (e) {
-      console.log("[debug]" + e);
-    }
+    await turnContext.updateActivity(updateCardActivity);
     return null;
```
Comment (wenytang-ms) [inline faq-plus/bot/common/adaptiveHelper.ts:38]: why have to return a null, if this return null can be removed?

File: faq-plus/bot/teamsbot.ts [hunk 2]
```diff
@@ -106,6 +121,50 @@ export class TeamsBot extends TeamsActivityHandler {
...
+    let text: string = "";
+
+    if (message.replyToId && message.value != null) {
+      text = this.ChangeStatus;
+    } else {
+      text = message.text?.toLowerCase()?.trim() ?? "";
+    }
+
+    try {
+      switch (text) {
```
Comment (SLdragon) [inline faq-plus/bot/teamsbot.ts:142]: If there are not other cases, you can move the switch logic to the above if sentence

File: faq-plus/bot/teamsbot.ts [hunk 3]
```diff
@@ -106,6 +121,50 @@ export class TeamsBot extends TeamsActivityHandler {
...
+
+        default:
+          console.log("Unrecognized input in channel");
+          await turnContext.sendActivity(
+            "todo: Unrecognized input in channel."
+          );
+          break;
+      }
+    } catch (error) {
+      // todo: Check if expert user is trying to delete the question and knowledge base has not published yet.
```
Comment (blackchoey) [inline faq-plus/bot/teamsbot.ts:156]: what feature we want to deliver with this todo?

File: faq-plus/bot/teamsbot.ts [hunk 4]
```diff
@@ -171,76 +266,68 @@ export class TeamsBot extends TeamsActivityHandler {
...
+          conversationParameter,
+          async (turnContext) => {
+            let activity = turnContext.activity;
+            const conversationResourceResponse: ConversationResourceResponse = {
+              id: activity.conversation.id,
+              activityId: activity.id,
+              serviceUrl: activity.serviceUrl,
+            };
+            resolve(conversationResourceResponse);
+            return Promise.resolve();
```
Comment (SLdragon) [inline faq-plus/bot/teamsbot.ts:317]: Why resolve multiple times?

Comment (blackchoey) [inline faq-plus/bot/teamsbot.ts:317]: there's already a `resolve()` statement above, do we need to to return Promise.resolve() here?

Comment (wenytang-ms) [reply faq-plus/bot/teamsbot.ts:317]: it seems this is an anti-pattern as follow
https://stackoverflow.com/questions/43036229/is-it-an-anti-pattern-to-use-async-await-inside-of-a-new-promise-constructor

File: faq-plus/bot/common/adaptiveHelper.ts (hunk 1 above)
Comment (dilin-MS) [reply faq-plus/bot/common/adaptiveHelper.ts:38]: When `askAnExpertSubmitTextPayload.Title` not exists, we does not create ticket and return null. If `askAnExpertSubmitTextPayload.Title` exists, we will create a ticket and return the ticket as a promise.

File: faq-plus/bot/teamsbot.ts (hunk 2 above)
Comment (dilin-MS) [reply faq-plus/bot/teamsbot.ts:142]: There may be other cases we have not yet added.

File: faq-plus/bot/teamsbot.ts (hunk 3 above)
Comment (dilin-MS) [reply faq-plus/bot/teamsbot.ts:156]: This remains to be decided.

Comment (blackchoey) [reply faq-plus/bot/teamsbot.ts:156]: can you share the behavior of existing FAQ Plus app? If it's not related to features we're going to deliver, we may update the logic to just throw an error and remove the todo.

File: faq-plus/bot/teamsbot.ts (hunk 4 above)
Comment (dilin-MS) [reply faq-plus/bot/teamsbot.ts:317]: fixed.

Comment (dilin-MS) [reply faq-plus/bot/teamsbot.ts:317]: removed.

File: faq-plus/bot/teamsbot.ts (hunk 3 above)
Comment (dilin-MS) [reply faq-plus/bot/teamsbot.ts:156]: See this [link](https://github.com/OfficeDev/microsoft-teams-apps-faqplus/blob/b128b7a4a88b199344c96d7318102c6bd10e08b1/Source/Microsoft.Teams.Apps.FAQPlusPlus/Bots/FaqPlusPlusBot.cs#L603) for code details.
If expert user try to delete question and knowledge base has not published yet, the error type will have bad request status code, and will show message to user to let them wait a while and try delete again: `Please wait for some time, updates to this question: "{0}" will be available in short time.`
I'm not sure whether we would support delete or publish command?

**Review by wenytang-ms — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Pull request merged**